rct229/rules/section22/section22rule9.py

```diff
@@ -0,0 +1,131 @@
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+
+        primary_secondary_loop_dictionary = get_primary_secondary_loops_dict(rmi_b)
+
+        return (
+            any(
+                [
+                    available_type in APPLICABLE_SYS_TYPES
+                    for available_type in available_type_lists
+                ]
+            )
+            and primary_secondary_loop_dictionary
+        )
+
+    def create_data(self, context, data):
+        rmr_baseline = context.baseline
+        rmi_b = rmr_baseline["ruleset_model_instances"][0]
+
+        chw_loop_capacity_dict = {}
+        for chiller in find_all("chillers[*]", rmi_b):
+            if chiller["cooling_loop"] not in chw_loop_capacity_dict.keys():
+                chw_loop_capacity_dict[chiller["cooling_loop"]] = ZERO.POWER
+            chw_loop_capacity_dict[chiller["cooling_loop"]] += getattr_(
+                chiller, "chiller", "rated_capacity"
+            )
+
+        primary_secondary_loop_dictionary = get_primary_secondary_loops_dict(rmi_b)
+
+        return {
+            "chw_loop_capacity_dict": chw_loop_capacity_dict,
+            "primary_secondary_loop_dictionary": primary_secondary_loop_dictionary,
+        }
+
+    def list_filter(self, context_item, data):
+        fluid_loop_b = context_item.baseline
+        primary_secondary_loop_dictionary = data["primary_secondary_loop_dictionary"]
+        chw_loop_capacity_dict = data["chw_loop_capacity_dict"]
+
+        return (
+            fluid_loop_b["id"] in primary_secondary_loop_dictionary.keys()
+            and chw_loop_capacity_dict[fluid_loop_b["id"]]
+            >= MIN_CHW_PRIMARY_LOOP_COOLING_CAPACITY
+        )
+
+    class ChillerFluidLoopRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section22Rule9.ChillerFluidLoopRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section22Rule9.ChillerFluidLoopRule.SecondaryChildLoopRule(),
+                index_rmr="baseline",
+                list_path="$..child_loops[*]",
+            )
+
+        class SecondaryChildLoopRule(RuleDefinitionBase):
+            def __init__(self):
+                super(
+                    Section22Rule9.ChillerFluidLoopRule.SecondaryChildLoopRule, self
+                ).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    required_fields={
+                        "$": ["cooling_or_condensing_design_and_control"],
+                        "cooling_or_condensing_design_and_control": [
+                            "minimum_flow_fraction"
+                        ],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                child_loop_b = context.baseline
+                secondary_loop_min_flow_rate = child_loop_b[
```

Comment:
Should this be called `secondary_loop_min_flow_fraction`?  Actually, I think `min_flow_fraction` may work better in the output since it will be part of the output for the secondary loop.

Reply:
Fixed. Great point!